crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -117,6 +116,36 @@ impl Analyzer<'_, '_> {
             | Type::EnumVariant(..)
             | Type::Param(_)
             | Type::Module(_) => return Ok(ty),
+            Type::Interface(Interface {
+                span,
+                name,
+                type_params,
+                extends,
+                body,
+                metadata: InterfaceMetadata { common },
+                tracker,
+            }) => {
+                for extend in extends {
+                    let types = extend.to_owned().type_args.into_iter().flat_map(|cur| cur.params);
+                    let union_type = Type::new_union(*span, types);
+                    if let box RExpr::Ident(ident) = &extend.expr {
```

Comment:
I think we should check for `Type::Array` after using `self.type_of_ts_entity_name`